apps/src/javalab/JavalabEditorTabMenu.jsx

```diff
@@ -48,14 +52,43 @@ class JavalabTabMenuComponent extends Component {
         >
           Delete
         </button>
-        {showVisibilityOption && (
+        {showVisibilityOption && !fileIsVisible && (
           <button
             type="button"
             key="visibility"
-            onClick={changeVisibilityFromTabMenu}
+            onClick={() => {
```

Comment:
since this is now two lines that are somewhat repeated across each option it would be good to extract it out into a helper method. It could be one method that's parameterized by isVisible and isValidation. If you go with that option, it would be good to add inline comments when calling it so it it's clear what each call is doing, such as (don't have to use this method name): `changeFileMetadata(true /*isVisible*/, false /*isValidation*/)`

Reply:
Updated- the formatting spec made me separate it out by line still, but I agree it's better than duplicated code!